### PR TITLE
changed nginx-rtmp fork and updated alpine to latest

### PR DESCRIPTION
**Pull request by reedog117**
Addresses #42 and #17 by changing compilation flags and utilizing a newer fork of the `nginx-rtmp` module.

### CODE REVIEW

**Comment by reedog117** [comment]
Added an extra compilation flag for `nginx` for a successful compile

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ARG NGINX_VERSION=1.16.1
-ARG NGINX_RTMP_VERSION=1.2.1
+ARG NGINX_RTMP_VERSION=2.0.0.6
 ARG FFMPEG_VERSION=4.2.1
 
 
 ##############################
 # Build the NGINX-build image.
-FROM alpine:3.8 as build-nginx
+FROM alpine:latest as build-nginx
```
Comment (alfg) [inline Dockerfile]: Can we pin this to `1.11` instead of `latest`?

File: Dockerfile [hunk 2]
```diff
@@ -36,7 +36,7 @@ RUN cd /tmp && \
 
 # Get nginx-rtmp module.
 RUN cd /tmp && \
-  wget https://github.com/arut/nginx-rtmp-module/archive/v${NGINX_RTMP_VERSION}.tar.gz && \
+  wget https://github.com/chetandhembre/nginx-rtmp-module/archive/v${NGINX_RTMP_VERSION}.tar.gz && \
```
Comment (alfg) [inline Dockerfile]: I like the idea of using a forked project that is actively maintained. However, I want to keep the `latest` tag on this project true to the original `nginx-rtmp-module` source. Can we instead create a new `Dockerfile.chetandhembre` file where I can build and push this as a separate tag?

**Comment by dromer** [comment]
@reedog117 I tried your branch, but it seems that the hls endpoints are trying to use curl. I installed this from apk and then get a nice flood from curl on stdout:

```
nginx-rtmp_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
nginx-rtmp_1  |                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
nginx-rtmp_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time2020/03/09 17:15:35 [notice] 1#1: signal 17 (SIGCHLD) received from 157
nginx-rtmp_1  |   Curren t2020/03/09 17:15:35 [notice] 1#1: unknown process 157 exited with code 7
nginx-rtmp_1  |  
nginx-rtmp_1  | %   To t a l        %  Re ce i v ed   %   X fe r d    Av e r ag e   Sp e e d  D  lToiamed     U Tpilmoea d       TiTmoet a Clur r e nStp
nginx-rtmp_1  | e n t         L e f t    S p e e d
     0          0        0  D l  o a0d     U p0l o a d     0T o t a l    0  S p e n t  0  - - :L-e-f:t- -  Sp-e-e:d--
- - :0- - : - -  0       0 0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
     0           0        D0l o a d    0U  p l o0a d       0T  o t a l  0    S p e n t0   - - :L-e-f:t- -  -S-p:e-e-d:
- : -0- : - -    0      0 
nginx-rtmp_1  | 0     0  c  0u r l  : 0   ( 7 )   F0a i l e d   t0o  -c-o:n-n-e:c-t-  t-o- :l-o-c:a-l-h o-s-t: -p-o:r-t-  8 0 8 0 0: Connection refused
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
nginx-rtmp_1  |   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: signal 17 (SIGCHLD) received from 161
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: unknown process 159 exited with code 7
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: unknown process 161 exited with code 7
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: unknown process 163 exited with code 7
nginx-rtmp_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
nginx-rtmp_1  |                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: signal 17 (SIGCHLD) received from 165
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: unknown process 165 exited with code 7
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 hls: close fragment n=0, client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 6#6: signal 17 (SIGCHLD) received from 171
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 hls: ngx_rtmp_mpegts_close_file /opt/data/hls/show_720p2628kbs/1583774131425.ts, client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
nginx-rtmp_1  |                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     02020/03/09 17:15:35 [notice] 6#6: signal 17 (SIGCHLD) received from 173
  0     0    2020/03/09 17:15:35 [error] 6#6: *4 Akamia: '/opt/data/hls/show_720p2628kbs/video.m3u8', client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  | 0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: signal 17 (SIGCHLD) received from 172
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 1#1: unknown process 172 exited with code 7
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 6#6: signal 17 (SIGCHLD) received from 176
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 Akamia: '/opt/data/hls/show_720p2628kbs/manifest.m3u8', client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 hls: ngx_rtmp_hls_write_variant_playlist file '/opt/data/hls/manifest.m3u8' (2: No such file or directory), client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 6#6: signal 17 (SIGCHLD) received from 178
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 S3: '/opt/data/hls/manifest.m3u8', client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  | 2020/03/09 17:15:35 [notice] 6#6: signal 17 (SIGCHLD) received from 180
nginx-rtmp_1  | 2020/03/09 17:15:35 [error] 6#6: *4 S3: '/opt/data/hls/video.m3u8', client: 127.0.0.1, server: 0.0.0.0:1935
nginx-rtmp_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
nginx-rtmp_1  |                                  Dload  Upload   Total   Spent    Left  Speed
  2020/03/09 17:15:35 [error] 6#6: *5 hls: close fragment n=0, client: 127.0.0.1, server: 0.0.0.0:1935
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0--:--:--     0
nginx-rtmp_1  | curl: (7) Failed to connect to localhost port 8080: Connection refused
```

.. etc.

**Comment by dromer** [comment]
Also tried to update just the current repo with newer alpine (3.11) and it fails compiling on this:

```
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g   -I src/core -I src/event -I src/event/modules -I src/os/unix -I /tmp/nginx-rtmp-module-1.2.1 -I objs -I src/http -I src/http/modules \
	-o objs/addon/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.o \
	/tmp/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.c
/tmp/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.c: In function 'ngx_rtmp_eval':
/tmp/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.c:160:17: error: this statement may fall through [-Werror=implicit-fallthrough=]
  160 |                 switch (c) {
      |                 ^~~~~~
/tmp/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.c:170:13: note: here
  170 |             case ESCAPE:
      |             ^~~~
cc1: all warnings being treated as errors
make[1]: *** [objs/Makefile:1395: objs/addon/nginx-rtmp-module-1.2.1/ngx_rtmp_eval.o] Error 1
make[1]: Leaving directory '/tmp/nginx-1.16.1'
make: *** [Makefile:8: build] Error 2
```

Which is probably due to newer gcc.

File: Dockerfile (hunk 1 above)
Comment (reedog117) [reply Dockerfile]: done

**Comment by reedog117** [comment]
hmm i haven't tried the other alternative modules in depth yet, although i've changed the Dockerfile back to use the original module. Let me experiment a little bit more.

**Comment by absentio** [comment]
What's better using this fork of nginx-rtmp?

**Comment by alfg** [comment]
Hi @reedog117, is this still being worked on?

**Comment by alfg** [comment]
Closing. Feel free to re-open if you wish to continue working on this.

**Pull request closed without merging**